fuel-vm/src/interpreter/memory.rs

```diff
@@ -403,6 +418,37 @@ impl MemoryInstance {
         &self.heap
     }
 
+    /// Diff of from `self` to `new` memory state.
+    /// Panics if new instance is not possible to reach from the current one,
+    /// for instance if it has smaller stack or heap allocation.
+    pub fn diff_patches(&self, new: &MemoryInstance) -> Vec<MemorySliceChange> {
+        let mut changes = Vec::new();
+        let mut current_change: Option<MemorySliceChange> = None;
+
+        for i in 0..MEM_SIZE {
```

Comment:
Ohh, it will be very expensive to do 10^7 iterations per each instruction=D It would be nice to have something more performant=)

If we go for now with `Tracer` approach, we can decide later in the `fuel-core` how do we want to do that in an optimal way

Reply:
I'm very aware of this. I have lots of ideas on how to optimize this, and I'm going to write some benchmarks to see what approach is the right one.